src/Compilers/CSharp/Portable/Symbols/Source/GlobalExpressionVariable.cs

```diff
@@ -73,12 +73,21 @@ internal override TypeSymbol GetFieldType(ConsList<FieldSymbol> fieldsBeingBound
 
             var diagnostics = DiagnosticBag.GetInstance();
             TypeSymbol type;
+            bool isVar;
 
             var binderFactory = compilation.GetBinderFactory(SyntaxTree);
-            var binder = binderFactory.GetBinder(typeSyntax);
+            var binder = binderFactory.GetBinder(typeSyntax ?? SyntaxNode);
```

Comment:
Consider folding this null test `??` into the null test below.

Reply:
Can't; the binder is used later, e.g. `binder.CreateErrorType(...)`

---
In reply to: [155099518](https://github.com/dotnet/roslyn/pull/23209#discussion_r155099518) [](ancestors = 155099518)